### PR TITLE
fix(ieee): Restore author.ts

### PR DESCRIPTION
**Pull request by Derekmini**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/ieee/author/37264968900/newest
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/routes/ieee/author.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    },
+    example: '/ieee/author/37264968900/newest/20',
+    handler,
+};
+
+async function handler(ctx) {
+    const { aid, sortType, count = 10 } = ctx.req.param();
+    const host = 'https://ieeexplore.ieee.org';
+
+    const author = await ofetch(`${host}/rest/author/${aid}`).then((res) => res[0]);
```
Comment (TonyRL) [inline lib/routes/ieee/author.ts:35]: Please [cache](https://docs.rsshub.app/joinus/advanced/use-cache) the response

File: lib/routes/ieee/author.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            })
+        )
+    );
+
+    return {
+        title,
+        link,
+        description,
+        item: items,
+    };
```
Comment (TonyRL) [inline lib/routes/ieee/author.ts:86]: Please add the `image` from `author.photoUrl`. The baseUrl is `https://ieeexplore.ieee.org/`

File: lib/routes/ieee/author.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            rowsPerPage: count,
+            searchWithin: [`"Author Ids": ${aid}`],
+            sortType,
+        },
+    });
+
+    const list = response.records.map((item) => {
+        const $ = load(item.articleTitle);
+        return {
+            title: $.text(),
```
Comment (TonyRL) [inline lib/routes/ieee/author.ts:52]: `articleTitle` is a plaintext only which does not require cheerio
```suggestion
        return {
            title: item.articleTitle,
```

File: lib/routes/ieee/author.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+                    const $ = load(res);
+                    const metadataMatch = $.html().match(/metadata=(.*);/);
+                    const metadata = metadataMatch ? JSON.parse(metadataMatch[1]) : null;
+                    const $2 = load(metadata?.abstract || '');
+                    item.abstract = $2.text();
+                }
+                return {
+                    ...item,
+                    description: renderDescription(item),
+                };
```
Comment (TonyRL) [inline lib/routes/ieee/author.ts:76]: `pubDate` is missing

File: lib/routes/ieee/author.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    return {
+        title,
+        link,
+        description,
+        item: items,
+    };
+}
+
+function renderDescription(item: { title: string; authors: string; abstract: string; doi: string }) {
+    const __dirname = getCurrentPath(import.meta.url);
```
Comment (TonyRL) [inline lib/routes/ieee/author.ts:90]: Please define __dirname in outer scope

File: lib/routes/ieee/author.ts [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+    const author = await ofetch(`${host}/rest/author/${aid}`).then((res) => res[0]);
+    const title = `${author.preferredName} on IEEE Xplore`;
+    const link = `${host}/author/${aid}`;
+    const description = author.bioParagraphs.join('<br/>');
+
+    const response = await ofetch(`${host}/rest/search`, {
+        method: 'POST',
+        body: {
+            rowsPerPage: count,
```
Comment (TonyRL) [inline lib/routes/ieee/author.ts:43]: Please make use of the [common parameter limit](https://docs.rsshub.app/guide/parameters#limit-entries) `ctx.req.query('limit')` instead.

File: lib/routes/ieee/author.ts [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+                    description: renderDescription(item),
+                };
+            })
+        )
+    );
+
+    return {
+        title,
+        link,
+        description,
```
[the remark this answers is not in the document]
Comment (Derekmini) [reply lib/routes/ieee/author.ts:85]: const description = author.bioParagraphs.join('\n');

File: lib/routes/ieee/author.ts (hunk 1 above)
Comment (Derekmini) [reply lib/routes/ieee/author.ts]:
```javescript
    const res = await ofetch(`${host}/rest/author/${aid}`);
    const author = res[0];
    const title = `${author.preferredName} on IEEE Xplore`;
```

File: lib/routes/ieee/author.ts (hunk 2 above)
Comment (Derekmini) [reply lib/routes/ieee/author.ts]:
```
const image = `${host}${author.photoUrl}`;


    return {
        title,
        link,
        description,
        item: items,
        image,
    };
```

File: lib/routes/ieee/author.ts (hunk 3 above)
Comment (Derekmini) [reply lib/routes/ieee/author.ts]:
```
    const list = response.records.map((item) => ({
        title: item.articleTitle,
        link: item.htmlLink,
        doi: item.doi,
        authors: 'authors' in item ? item.authors.map((itemAuth) => itemAuth.preferredName).join('; ') : 'Do not have author',
        abstract: 'abstract' in item ? item.abstract : '',
    }));
```

File: lib/routes/ieee/author.ts (hunk 4 above)
Comment (Derekmini) [reply lib/routes/ieee/author.ts:77]:
```
                    item.pubDate = metadata?.insertDate ? parseDate(metadata.insertDate) : undefined;
                    item.abstract = load(metadata?.abstract || '').text();
                }
                return {
                    ...item,
                    description: renderDescription(item),
                };
```

File: lib/routes/ieee/author.ts (hunk 5 above)
Comment (Derekmini) [reply lib/routes/ieee/author.ts]:
```
import { getCurrentPath } from '@/utils/helpers';
const __dirname = getCurrentPath(import.meta.url);
```

```
function renderDescription(item: { title: string; authors: string; abstract: string; doi: string }) {
    return art(path.join(__dirname, 'templates/description.art'), {
        item,
    });
}
```

File: lib/routes/ieee/author.ts [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+async function handler(ctx) {
+    const { aid, sortType } = ctx.req.param();
+    const count = ctx.req.query('limit') || 10;
+    const host = 'https://ieeexplore.ieee.org';
+
+    const res = await ofetch(`${host}/rest/author/${aid}`);
+    const author = res[0];
+    const title = `${author.preferredName} on IEEE Xplore`;
+    const link = `${host}/author/${aid}`;
+    const description = author.bioParagraphs.join('\n');
```
Comment (TonyRL) [inline lib/routes/ieee/author.ts:41]:
```suggestion
    const description = author.bioParagraphs.join(' ');
```

**Pull request merged**